packages/cicero-core/src/instanceloader.js

```diff
@@ -0,0 +1,205 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const JSZip = require('jszip');
+
+const Logger = require('@accordproject/concerto-core').Logger;
+const FileLoader = require('@accordproject/ergo-compiler').FileLoader;
+const LogicManager = require('@accordproject/ergo-compiler').LogicManager;
+const ParserManager = require('@accordproject/markdown-template').ParserManager;
+const CiceroMarkTransformer = require('@accordproject/markdown-cicero').CiceroMarkTransformer;
+const ErgoEngine = require('@accordproject/ergo-engine/index.browser.js').EvalEngine;
+
+const InstanceMetadata = require('./instancemetadata');
+const Util = require('./util');
+
+/**
+ * A utility class to create smart legal contract instances from data sources.
+ * @class
+ * @private
+ * @abstract
+ */
+class InstanceLoader extends FileLoader {
+    /**
+     * Create an instance from a Template and data.
+     * @param {*} Instance - the type to construct
+     * @param {Template} template  - the template for the instance
+     * @param {object} data - the contract data
+     * @param {string} instantiator - name of the person/party which instantiates the contract instance
+     * @return {object} - the contract instance
+     */
+    static fromTemplateWithData(Instance, template, data, instantiator) {
+        const metadata = InstanceMetadata.createMetadataFromTemplate(template.getMetadata());
+        const logicManager = template.getLogicManager();
+        const grammar = template.getParserManager().getTemplate();
+
+        // create the instance
+        const instance = new (Function.prototype.bind.call(
+            Instance,
+            null,
+            metadata,
+            logicManager,
+            grammar,
+            template,
+            instantiator
+        ));
+
+        instance.setData(data);
+
+        //grab the parties
+        const contractModel = Util.getContractModel(instance.logicManager, instance.instanceKind);
+        const properties = contractModel.getProperties();
+        properties.map((property) => property.getDecorators().map((decorator) => {
+            if (decorator.getName() === 'ContractParty') {
```

Comment:
How is the Decorator used? It looks like any property on the template model can be decorated as the Party name? Why are we not using the `org.accordproject.Party` type?

Reply:
Because we might extend (in fact, we do extend) the `participant` (e.g. `participant AgreementParty extends Party`) and so we need to know the specific `concept` to be designated as a signatory.